docs/PLUGIN_DIAGNOSTICS.md

```diff
@@ -0,0 +1,67 @@
+# Plugin Diagnostics
+
+Plugin diagnostics provides a simple way to verify that a plugin is capable of running without requiring the plugin to be loaded and a task started.
+This feature works for collector plugins written using one of our snap-plugin-libs ([snap-plugin-lib-go](https://github.com/intelsdi-x/snap-plugin-lib-go), [snap-plugin-lib-py](https://github.com/intelsdi-x/snap-plugin-lib-py), [snap-plugin-lib-cpp](https://github.com/intelsdi-x/snap-plugin-lib-cpp)).
+
+Plugin diagnostics delivers following information:
+- runtime details (plugin name, version, etc.)
+- configuration details
+- list of metrics exposed by the plugin
+- metrics with their values that can be collected right now
+- and times required to retrieve this information
+
+## Running plugin diagnostics
+It is possible to run plugin diagnostics without any arguments (default values are used), e.g.:
+```
+$ ./snap-plugin-collector-psutil
+```
+or with plugin configuration in the form of a JSON:
+```
+$ ./snap-plugin-collector-psutil --config {\"mouting_points\":\"/dev/sda\"}
```

Comment:
@katarzyna-z, could you check the way how to provide the config? Ssee my example below, but please verify that. The another thing is that the psutil collector does not have `mouting_points` config param, but ` mount_points`, so it's better to use config's name which really exists

```
$ ./snap-plugin-collector-psutil --config '{"mount_points":"/dev/sda"}'
```